core/src/main/java/org/elasticsearch/search/aggregations/bucket/range/ipv4/IPv4RangeBuilder.java

```diff
@@ -112,55 +117,69 @@ public IPv4RangeBuilder addUnboundedFrom(String from) {
     /**
      * Computes the min &amp; max ip addresses (represented as long values - same way as stored in index) represented by the given CIDR mask
      * expression. The returned array has the length of 2, where the first entry represents the {@code min} address and the second the {@code max}.
-     * A {@code -1} value for either the {@code min} or the {@code max}, represents an unbounded end. In other words:
-     *
-     * <p>
-     * {@code min == -1 == "0.0.0.0" }
-     * </p>
-     *
-     * and
-     *
-     * <p>
-     * {@code max == -1 == "255.255.255.255" }
-     * </p>
      */
     static long[] cidrMaskToMinMax(String cidr) {
-        String[] parts = MASK_PATTERN.split(cidr);
-        if (parts.length != 5) {
-            return null;
+        Objects.requireNonNull(cidr, "cidr");
+        String[] fields = cidr.split("/");
+        if (fields.length != 2) {
+            throw new IllegalArgumentException(
+                    String.format(Locale.ROOT, "expected [a.b.c.d, e] but was [%s] after splitting on \"/\" in [%s]", Arrays.toString(fields), cidr)
+            );
         }
-        int addr = (( Integer.parseInt(parts[0]) << 24 ) & 0xFF000000)
-                | (( Integer.parseInt(parts[1]) << 16 ) & 0xFF0000)
-                | (( Integer.parseInt(parts[2]) << 8 ) & 0xFF00)
-                |  ( Integer.parseInt(parts[3]) & 0xFF);
-
-        int mask = (-1) << (32 - Integer.parseInt(parts[4]));
-
-        if (Integer.parseInt(parts[4]) == 0) {
-            mask = 0 << 32;
+        String[] ipAddressFields = fields[0].split("\\.");
+        if (ipAddressFields.length != 4) {
+            throw new IllegalArgumentException(
```

Comment:
I've added a note in the Javadoc that this method throws IllegalArgumentException in 45ab5dd6eabdcbcd7dd0d2b68867438e99f22f8e. Either the caller correctly catches these (and just wraps them in one of our fancy exceptions) or the caller doesn't catch these and we still blow up. I see this as no different than the way that NumberFormatException is handled, for example, in the JDK.